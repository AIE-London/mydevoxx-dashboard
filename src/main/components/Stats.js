/**
 * Created by DLINDSAY on 05-Apr-17.
 */
import { Row, Col } from "react-flexbox-grid";
import Card from "./Card";
import styled from "styled-components";
import { CommaList, CommaListItem } from "./CommaList";
import React, { Component } from "react";

const Container = styled.div`
  color: #000;
  height: 100%;
  margin-bottom: 1em;
  max-width: 520px;
  height: 100%;
`;

const StatsCard = styled(Card)`
  height: 100%;
`;

const StatHeader = styled.h2`
    opacity: 0.8;
    font-family: helvetica;
`;

class Stats extends Component {
  render() {
    try {
      return (
        <Container>
          <StatsCard center="xs">
            {/* Wraps the Card component within the Container component */}
            <Col xs={11}>
              <Col xs={6}>
                <Row start="xs">
                  <h1>My Stats</h1>
                </Row>
              </Col>

              <Row start="xs">
                <Col xs={12}>
                  <StatHeader>Top Tracks:</StatHeader>
                  <div style={{ textAlign: "center" }}>
                    <CommaList>
                      {/* creates a ul element using the CommaList styled ul format */}
                      {this.props.tracks.map(function(track) {
                        return (
                          <CommaListItem key={track}>{track}</CommaListItem>
                        );
                      })}
                    </CommaList>
                  </div>

                  <StatHeader>Top Speakers:</StatHeader>

                  <div style={{ textAlign: "center", paddingBottom: 15 }}>
                    <CommaList>
                      {this.props.speakers.map(function(speaker) {
                        return (
                          <CommaListItem key={speaker}>{speaker}</CommaListItem>
                        );
                      })}
                    </CommaList>
                  </div>
                </Col>
              </Row>
            </Col>

          </StatsCard>
        </Container>
      );
    } catch (error) {
      return (
        <Container>
          <StatsCard center="xs">
            {/* Wraps the Card component within the Container component */}
            <Col xs={11}>
              <Col xs={6}>
                <Row start="xs">
                  <h1>My Stats</h1>
                </Row>
              </Col>

              <Row start="xs">
                <Col xs={12}>
                  <StatHeader>Top Tracks:</StatHeader>

                  <StatHeader>Top Speakers:</StatHeader>
                </Col>
              </Row>
            </Col>

          </StatsCard>
        </Container>
      );
    }
  }
}

export default Stats;
