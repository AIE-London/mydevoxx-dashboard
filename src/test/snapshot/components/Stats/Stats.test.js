import React from "react";
import ReactDOM from "react-dom";
import Stats from "../../../../main/components/Stats";
import renderer from "react-test-renderer";

test("card component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Stats
        tracks={["Agile", "Bgile", "Cgile"]}
        speakers={["Person One", "Person Two", "Person Three"]}
      />,
      div
    );
  });
});
test("stats component with all properties filled", () => {
  const tree = renderer
    .create(
      <Stats
        tracks={["Agile", "Bgile", "Cgile"]}
        speakers={["Person One", "Person Two", "Person Three"]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
