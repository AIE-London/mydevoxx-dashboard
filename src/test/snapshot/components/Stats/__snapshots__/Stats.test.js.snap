exports[`test stats component with all properties filled 1`] = `
<div
  className="dRQkYN">
  <div
    className="bvnlUe row center-xs">
     
    <div
      className="col-xs-10">
       
      <div
        className="col-xs-6">
         
        <div
          className="row start-xs">
           
          <h1>
            My Stats
          </h1>
        </div>
      </div>
      <div
        className="row start-xs">
         
        <div
          className="col-xs-12">
           
          <h2
            className="eIkKGg">
            Top Tracks:
          </h2>
           
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }>
             
            <ul
              className="hJKnQv">
               
              <li
                className="cLQKgI">
                Agile
              </li>
              <li
                className="cLQKgI">
                Bgile
              </li>
              <li
                className="cLQKgI">
                Cgile
              </li>
            </ul>
          </div>
          <h2
            className="eIkKGg">
            Top Speakers:
          </h2>
           
          <div
            style={
              Object {
                "paddingBottom": 15,
                "textAlign": "center",
              }
            }>
             
            <ul
              className="hJKnQv">
               
              <li
                className="cLQKgI">
                Person One
              </li>
              <li
                className="cLQKgI">
                Person Two
              </li>
              <li
                className="cLQKgI">
                Person Three
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
